File: app/assets/javascripts/models/post.js
Labs.Post = Labs.Commentable.extend({

  content: DS.attr("string"),

  title: DS.attr('string'),
  description: DS.attr('string'),
  original_url: DS.attr('string'),
  thumbnail_url: DS.attr('string'),
  html: DS.attr('string'),

  project: DS.belongsTo("Labs.Project")

});

File: app/models/post.rb
class Post < ActiveRecord::Base

  belongs_to :project
  belongs_to :user

  has_many :comments, :as => :commentable

  validates :content, :presence => true
  validates :user, :presence => true

  store :metadata,
    :accessors => [:title, :description, :original_url, :thumbnail_url, :html],
    :coder => JSON

end
